src/hooks/usePagination/usePagination.ts

```diff
@@ -6,7 +6,25 @@ const arraysAreEqual = <Q>(arr1: Q[], arr2: Q[]) =>
         (value, index) => JSON.stringify(value) === JSON.stringify(arr2[index])
     );
 
-export const usePagination = <T>(
+const updateData = <T extends { id: string | number }>(
+    prevData: T[],
+    newData: T[]
+): T[] => {
+    const updatedData = [...prevData];
+    newData.forEach(newItem => {
```

Comment:
Maybe we can use a map here for indexes of updated data to improve performance. findIndex will run every element of newData, this is O(n^2) which will not be very efficient for large amount of data

Reply:
done